samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -331,9 +332,24 @@ public void launchStreamProcessor(SamzaResource resource, CommandBuilder builder
   public void stopStreamProcessor(SamzaResource resource) {
     synchronized (lock) {
       Container container = allocatedResources.get(resource);
+      /*
+       * 1. Stop the container through NMClient if the container was instantiated as part of NMClient lifecycle.
+       * 2. Stop the container through AMClient by release the assigned container if the container was from the previous
+       *    attempt and managed by the AM due to AM-HA
+       * 3. Ignore the request if the container associated with the resource isn't present in the bookeeping.
+       */
       if (container != null) {
         log.info("Stopping Container ID: {} on host: {}", resource.getContainerId(), resource.getHost());
         this.nmClientAsync.stopContainerAsync(container.getId(), container.getNodeId());
+      } else {
+        YarnContainer yarnContainer = state.runningProcessors.get(resource.getContainerId());
```

Comment:
major: hmm.. some confusion here.. resource.getContainerId() gets the yarn container id right (of the form `container_1350670447861_0003_01_000001`).  see one ex [here ](https://github.com/apache/samza/blob/master/samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java#L488)

But if we see code where the runningProcessors gets populated -- it has key as samza processorId (of the form 0) -- see [here ](https://github.com/apache/samza/blob/master/samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java#L738)

but here you are using container id to fetch from runningProcessors.

Reply:
Sync'd offline on this. It only uses processorId and the prototype was using a buggy version which populated it with containerId. Fixed it.

Thanks for calling this out 👍